File: src/Web.API/Models/Identity/Login/LoginResponseModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Threading.Tasks;

namespace Web.API.Models.Identity.Login;

public class LoginResponseModel
{
    public bool IsSucceeded { get; set; }
    public string Message { get; set; }

    public string UserId { get; set; }
    public string Username { get; set; }

    public string Token { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime ExpiresAt { get; set; }
    public string RefreshToken { get; set; }

    public LoginResponseModel(bool isSucceeded, string message)
    {
        IsSucceeded = isSucceeded;
        Message = message;
    }

    public LoginResponseModel(string userId, string username, string token, DateTime createdAt, DateTime expiresAt, string refreshToken)
    {
        IsSucceeded = true;

        UserId = userId;
        Username = username;
        Token = token;
        CreatedAt = createdAt;
        ExpiresAt = expiresAt;
        RefreshToken = refreshToken;
    }
}

File: src/Web.Angular/ClientApp/src/app/Models/Auth/LoginResponseModel.ts
import { CookieService } from "ngx-cookie-service";

export class LoginResponseModel {
  isSucceeded: boolean = false;
  message: string = '';
  userId: string = '';
  username: string = '';
  token: string = '';
  createdAt: string = '';
  expiresAt: string = '';
  refreshToken: string = '';

  constructor(
    isSucceeded: boolean,
    message: string,
    userId: string,
    username: string,
    token: string,
    createdAt: string,
    expiresAt: string,
    refreshToken: string,
    private cookieService:CookieService
  ) {
    this.isSucceeded = isSucceeded;
    this.message = message;
    this.userId = userId;
    this.username = username;
    this.token = token;
    this.createdAt = createdAt;
    this.expiresAt = expiresAt;
    this.refreshToken = refreshToken;
  }

  setToLocalStorage(){

    localStorage.setItem('userId', this.userId);
    localStorage.setItem('username', this.username);

    localStorage.setItem('token', this.token);
    localStorage.setItem('createdAt', this.createdAt);
    localStorage.setItem('expiresAt', this.expiresAt);
    localStorage.setItem('refreshToken', this.refreshToken);
  }

  setToCookies(){

    this.cookieService.set('userId', this.userId);
    this.cookieService.set('username', this.username);

    this.cookieService.set('token', this.token);
    this.cookieService.set('createdAt', this.createdAt);
    this.cookieService.set('expiresAt', this.expiresAt);
    this.cookieService.set('refreshToken', this.refreshToken);
  }
}
